x/feegrant/keeper/keeper.go

```diff
@@ -49,6 +50,25 @@ func (k Keeper) GrantAllowance(ctx sdk.Context, granter, grantee sdk.AccAddress,
 
 	store := ctx.KVStore(k.storeKey)
 	key := feegrant.FeeAllowanceKey(granter, grantee)
+
+	var exp *time.Time
+	existingGrant, err := k.getGrant(ctx, grantee, granter)
+	if err != nil && existingGrant.GetAllowance() != nil {
+		grant1, err := existingGrant.GetGrant()
+		if err != nil {
+			return err
+		}
+
+		exp, err = grant1.ExpiresAt()
+		if err != nil {
+			return err
+		}
+
+		if exp != nil {
+			k.removeFromGrantQueue(ctx, feegrant.FeeAllowancePrefixQueue(exp, key))
```

Comment:
shall we move `feegrant.FeeAllowancePrefixQueue` call into the `k.removeFromGrantQueue` body?